tower/src/discover/mod.rs

```diff
@@ -104,3 +104,16 @@ pub enum Change<K, V> {
     /// The service identified by key `K` disappeared.
     Remove(K),
 }
+
+impl<K, V> Clone for Change<K, V>
+where
+    K: Clone,
+    V: Clone,
+{
+    fn clone(&self) -> Self {
+        match self {
+            Change::Insert(k, v) => Change::Insert(k.clone(), v.clone()),
+            Change::Remove(k) => Change::Remove(k.clone()),
+        }
+    }
+}
```

Comment:
This implementation seems like it could be derived?

Reply:
Yep, you're right - I had some silly notions around deriving clone on enums with generics. Have fixed now, thanks!